crates/sail-common/src/spec/data_type.rs

```diff
@@ -88,28 +333,172 @@ pub struct Field {
     pub metadata: Vec<(String, String)>,
 }
 
+/// [Credit]: The implementations for [`Field`], [`FieldRef`], [`Fields`], and [`UnionFields`] are
+/// copied from [`arrow_schema::Field`], [`arrow_schema::FieldRef`], [`arrow_schema::Fields`], and [`arrow_schema::UnionFields`].
+///
+/// A reference counted [`Field`]
+pub type FieldRef = Arc<Field>;
```

Comment:
I wonder if we could keep our previous definition of `Field`, `Fields`, etc. The spec does not need to be shared across threads, so the overhead of `Arc` is probably unnecessary.